### PR TITLE
chore: adapt to aleo crypto

### PR DESCRIPTION
**Pull request by joske**

### CODE REVIEW

File: node/Cargo.toml [hunk 1]
```diff
@@ -51,9 +51,16 @@ features = ["sink"]
...
+[dependencies.narwhal-node]
+workspace = true
+
+[dependencies.narwhal-crypto]
+workspace = true
+
 [dependencies.narwhal-config]
 git = "https://github.com/eqlabs/bullshark-bft/"
 package = "narwhal-config"
+branch="crypto"
```
Comment (ljedrz) [inline node/Cargo.toml]:
```suggestion
branch = "crypto"
```

File: node/bft-consensus/src/lib.rs [hunk 2]
```diff
@@ -25,12 +25,16 @@ pub use validation::TransactionValidator;
 
 use anyhow::Result;
 use arc_swap::ArcSwap;
-use fastcrypto::{bls12381::min_sig::BLS12381KeyPair, traits::KeyPair};
 use multiaddr::Protocol;
 use narwhal_config::{Committee, Import, Parameters, WorkerCache};
-use narwhal_crypto::NetworkKeyPair;
+use narwhal_crypto::{KeyPair, NetworkKeyPair};
```
Comment (ljedrz) [inline node/bft-consensus/src/lib.rs]: I'd prefer `KeyPair` to be aliased as `NarwhalKeypair` for extra clarity (applies everywhere it's used)

File: node/messages/Cargo.toml [hunk 3]
```diff
@@ -46,6 +45,13 @@ version = "1"
 [dependencies.serde]
 version = "1"
 
+[dependencies.serde_json]
+version = "1"
+
+[dependencies.rand]
+version = "0.8.5"
+features = ["small_rng"]
```
Comment (ljedrz) [inline node/messages/Cargo.toml]: why are we using `small_rng` here? it should only be used in tests, as it's insecure

File: node/messages/src/consensus_id.rs [hunk 4]
```diff
@@ -32,15 +30,65 @@ impl MessageTrait for Box<ConsensusId> {
     }
 
     fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
-        writer.write_all(self.public_key.as_bytes())?;
-        writer.write_all(self.signature.as_bytes())?;
+        // bincode::serialize_into(writer, &(&self.public_key, &self.signature))?;
```
Comment (ljedrz) [inline node/messages/src/consensus_id.rs]: this should ideally be fixed in snarkVM first

File: node/messages/src/consensus_id.rs [hunk 5]
```diff
@@ -32,15 +30,65 @@ impl MessageTrait for Box<ConsensusId> {
...
+
+#[cfg(test)]
+mod test {
+    use bytes::BufMut;
+    use narwhal_crypto::KeyPair;
+
+    use super::*;
+
+    #[test]
+    fn test_consensus_id() {
```
Comment (ljedrz) [inline node/messages/src/consensus_id.rs]:
```suggestion
    fn consensus_id_serialization() {
```

File: node/messages/src/consensus_id.rs [hunk 6]
```diff
@@ -32,15 +30,65 @@ impl MessageTrait for Box<ConsensusId> {
...
+        let id = Box::new(ConsensusId { public_key: public.clone(), signature });
+        let mut buf = BytesMut::with_capacity(128).writer();
+        id.serialize(&mut buf).unwrap();
+        let bytes = buf.into_inner();
+        let deserialized = MessageTrait::deserialize(bytes).unwrap();
+        assert_eq!(id, deserialized);
+    }
+
+    #[test]
+    fn test_signature() {
```
Comment (ljedrz) [inline node/messages/src/consensus_id.rs]:
```suggestion
    fn signature_serialization() {
```

Comment (ljedrz) [reply node/messages/src/consensus_id.rs]: also, the right place for serialization tests of snarkVM objects is snarkVM, so these tests should ultimately be removed if they exist there, and added if not

**Review by ljedrz — APPROVED** [review]
LGTM in general, but we should likely wait until the branch in `AleoHQ/snarkOS` is blessed and apply it against it instead of merging it here

**Comment by niklaslong** [comment]
This one can probably be closed now right?

**Comment by joske** [comment]
this is merged

**Pull request closed without merging**